src/components/scores/late-work.cjsx

```diff
@@ -1,113 +1,120 @@
-React  = require 'react'
-Router = require 'react-router'
-BS = require 'react-bootstrap'
-Time   = require '../time'
+React      = require 'react'
+BS         = require 'react-bootstrap'
+Time       = require '../time'
+classnames = require 'classnames'
 
 {ScoresStore, ScoresActions} = require '../../flux/scores'
 
-LateWork = React.createClass
-  displayName: 'LateWork'
+class LateWork
+  constructor: (@task) ->
+    @isAccepted = @task.is_late_work_accepted
+    @status = if @isAccepted then 'accepted' else 'pending'
+
+    @displayValue =
+        if @isAccepted
+          @task.correct_on_time_exercise_count + @task.correct_accepted_late_exercise_count
+        else
+          @task.correct_on_time_exercise_count
 
-  setLateStatus: ->
-    {task, courseId, period_id, columnIndex, isIncludedInAverages, currentValue, acceptValue} = @props
-    isAccepted = task.is_late_work_accepted
-    if not @isUpdatingLateStatus()
-      if isAccepted
-        ScoresActions.rejectLate(task.id, courseId)
-      else
-        ScoresActions.acceptLate(task.id, courseId)
-
-      if isIncludedInAverages
-        updateAveragesParams = [
-          task,
-          courseId,
-          period_id,
-          columnIndex,
-          currentValue,
-          acceptValue
-        ]
-        ScoresActions.updateAverages(updateAveragesParams...)
+  lateExerciseCount: ->
+    @task.completed_exercise_count - @task.completed_on_time_exercise_count
 
-    @refs.overlay.hide()
+  timeDisplay: ->
+    if @isAccepted
+      'due date'
+    else
+      <Time date={@task.last_worked_at} format='shortest'/>
+
+  className: ->
+    classnames( 'late-work-info-popover', @keyword, {
+      accepted: @isAccepted
+    })
+
+  get: (type) ->
+    @[type]()[@status]
+
+class HomeworkContent extends LateWork
+  keyword:  'homework'
+
+  title: ->
+    accepted: "You accepted this student's late score."
+    pending:  "#{@lateExerciseCount()} questions worked after the due date"
+  button: ->
+    accepted: 'Use this score'
+    pending:  'Accept late score'
+
+class ReadingContent extends LateWork
+  keyword:  'reading'
+
+  title: ->
+    accepted: "You accepted this student's late reading progress."
+    pending:  "Reading progress after the due date"
+  button: ->
+    accepted: 'Use this score'
+    pending:  'Accept late score'
+
+LateWorkPopover = React.createClass
 
+  componentWillMount: ->
+    Content = if @props.task.type is 'homework' then HomeworkContent else ReadingContent
+    @setState(
+      content: new Content(@props.task)
+    )
 
-  isUpdatingLateStatus: ->
-    ScoresStore.isUpdatingLateStatus(@props.task.id)
+  onButtonClick: ->
+    if @state.content.isAccepted
```

Comment:
I think when you have the "Additional late work" popover, this check will not be sufficient (because you'll be in the `isAccepted == true` state but wanting to call `acceptLate` again)